src/applications/hca/containers/IDPage.jsx

```diff
@@ -6,26 +6,56 @@ import FormTitle from 'platform/forms-system/src/js/components/FormTitle';
 import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
 import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 
+import { setData } from 'platform/forms-system/src/js/actions';
+import { getNextPagePath } from 'platform/forms-system/src/js/routing';
 import { focusElement } from 'platform/utilities/ui';
 import { toggleLoginModal } from 'platform/site-wide/user-nav/actions';
-import { isProfileLoading } from 'platform/user/selectors';
+import { isLoggedIn, isProfileLoading } from 'platform/user/selectors';
 
 import IDForm from '../components/IDForm';
 
 import { submitIDForm } from '../actions';
 
 class IDPage extends React.Component {
   componentDidMount() {
+    // Redirect to intro if a logged in user navigated to this page
+    // from another form page.
+    if (this.props.shouldRedirect) this.props.router.push('/');
     focusElement('.va-nav-breadcrumbs-list');
   }
 
   componentDidUpdate() {
-    // there's no need for logged-in users to see this page
-    if (this.props.shouldHideIDForm) {
-      this.props.router.push('/');
+    const { enrollmentStatus, form, hasOptionalDD214Upload } = this.props;
+
+    // Redirect to intro if a logged in user directly accessed this page.
+    if (this.props.shouldRedirect) this.props.router.push('/');
```

Comment:
oh, duh, this is why the id-form route is skipped for logged-in users.

Reply:
But if this redirects back to the intro, how does it work for going from the intro to the veteran information form page? This only handles the case where you directly go to the `/id-form` URL. The `depends` callback doesn't handle the case of going directly to the page, I don't think, or maybe it does at a lower level, but I think it should work for navigating using the form's "continue" and "back" buttons.

Reply:
Maybe try seeing if `isActivePage` in `src/platform/forms-system/src/js/helpers.js` is what's actually making use of the `depends` callback. It might not be getting called directly, since it's used as an argument to Lodash's `match` function.